test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -749,6 +755,19 @@ private static Streamable tryCreateNewInstance(Streamable streamable) throws NoS
         }
     }
 
+    private static Streamable tryCreateFromStream(Streamable streamable, StreamInput in) throws NoSuchMethodException,
+        InstantiationException, IllegalAccessException, InvocationTargetException {
+        try {
+            Class<? extends Streamable> clazz = streamable.getClass();
+            Constructor<? extends Streamable> constructor = clazz.getConstructor(StreamInput.class);
+            assertThat(constructor, Matchers.notNullValue());
+            Streamable newInstance = constructor.newInstance(in);
+            return newInstance;
+        } catch (Exception e) {
```

Comment:
I think you should catch the `NoSuchMethodException` explicitly, otherwise this could mask an exception thrown from the ctor.

Reply:
good call. forgot about the good ol IOException